scripts/functions/vae.py

```diff
@@ -63,15 +24,61 @@
 from functions.helper_ae import sampling_maker, CustomVariationalLayer, WarmUpCallback
 
 
-def tybalt_2layer_model(learning_rate, batch_size, epochs, kappa, intermediate_dim,
-                        latent_dim, epsilon_std, base_dir, analysis_name):
+def tybalt_2layer_model(
+        learning_rate,
+        batch_size,
+        epochs,
+        kappa,
+        intermediate_dim,
+        latent_dim,
+        epsilon_std,
+        base_dir,
+        analysis_name):
     """
     Train 2-layer Tybalt model using input dataset
 
     Output:
      Encoding and decoding neural networks to use in downstream analysis
     """
 
+    # The below is necessary in Python 3.2.3 onwards to
+    # have reproducible behavior for certain hash-based operations.
+    # See these references for further details:
+    # https://docs.python.org/3.4/using/cmdline.html#envvar-PYTHONHASHSEED
+    # https://github.com/keras-team/keras/issues/2280#issuecomment-306959926
+    randomState = 123
+    import os
+    os.environ['PYTHONHASHSEED'] = '0'
```

Comment:
why do you need to set the environment variable here? (Personal curiosity)

Reply:
I definitely don't know all the details for this.  This was also based on a blog post that I found as I was looking for ways to make multiple VAE training runs reproducible.  In general VAE training is known to yield slightly different models, but this one post said that this seemed to work: https://keras.io/getting-started/faq/#how-can-i-obtain-reproducible-results-using-keras-during-development